src/editor/nodes/EditorNodeMixin.js

```diff
@@ -208,34 +208,73 @@ export default function EditorNodeMixin(Object3DClass) {
       }
     }
 
-    addGLTFComponent(name, props) {
-      if (!this.userData.gltfExtensions) {
-        this.userData.gltfExtensions = {};
-      }
+    /**
+     * Adds a GLTF component
+     * @param {String} name The component name to be replaced
+     * @param {Object} props The component properties to be set
+     * @param {boolean} params Parameters object
+     * @param {{THREE.Object3D} params.node The target Object3D element
+     * @param {boolean} params.replace Wether or not the component should be replaced or just added if it does not exist
+     * @param {boolean} params.recursive Call this method recursively on all the Object3D children
+     * @param {boolean} params.firstPass If this is the first execution of the method
+     */
+    addGLTFComponent(name, props, params = {}) {
```

Comment:
@keianhzo can you explain the need for adding components recursively? I'm not sure what this is doing.

Reply:
The recursive flag is used to traverse the node hierarchy. When used along with the replace flag, you make sure you update the node holding a specific component instead of just adding it to the top level node as we were doing.

The loop-animation component has to be added to the Armature node in Blender to work in Hubs so we need to traverse the node hierarchy in Spoke to replace it in the child Armature node, otherwise Hubs will just read the original loop-animation component in the Armature node and use that.